src/main/java/com/gmail/uramen66/ordermovies/service/HallService.java

```diff
@@ -22,16 +25,32 @@ public HallDTO createHall(HallDTO hallDTO){
         Hall saveHall = hallRepository.saveAndFlush(hall);
         return hallMapper.hallToHallDto(saveHall);
     }
-    public HallDTO findHallById(Long id) throws Exception{
+
+
+    public HallDTO findHallById(Long id) {
         return hallRepository
                 .findById(id)
                 .map(hallMapper::hallToHallDto)
-                .orElseThrow(Exception::new);
+                .orElseThrow(ResourceNotFoundException ::new);
```

Comment:
I suggested to throw ResourceNotFoundException with some additional description. Like "Hall by id is not present"